packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/shared/base-load-balancer.ts

```diff
@@ -250,7 +251,9 @@ export abstract class BaseLoadBalancer extends Resource {
       this.setAttribute('load_balancing.cross_zone.enabled', baseProps.crossZoneEnabled === true ? 'true' : 'false');
     }
 
-    if (baseProps.denyAllIgwTraffic !== undefined) {
+    if (additionalProps.ipAddressType === IpAddressType.IPV4 && baseProps.denyAllIgwTraffic === false) {
```

Comment:
`ipAddressType` is meant to be `IpAddressType.IPV4` by default, and `denyAllIgwTraffic` is supposed to be `false`, I assume this condition is incorrect?

Reply:
Appreciate the feedback! Apologies for the spiel

Yes, `ipAddressType` is meant to be `IpAddressType.IPV4`. As for `denyAllIgwTraffic` the docs mention [_"optional, default: false for internet-facing load balancers and true for internal load balancers"_](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_elasticloadbalancingv2.ApplicationLoadBalancer.html#denyalligwtraffic). Looking at code though, `denyAllIgwTraffic` is never actually given a default value which is why we have this check for `undefined` here.

The `denyAllIgwTraffic` was created from this issue: https://github.com/aws/aws-cdk/issues/29520. Basically 
`denyAllIgwTraffic` is supposed to set `ipv6.deny_all_igw_traffic`. However, having `ipv6.deny_all_igw_traffic` set (to either true or false) for _any_ Ipv4 load balancer will cause a failed deployment. I'm starting to think the condition should actually be this:

```suggestion
    if (additionalProps.ipAddressType === IpAddressType.IPV4 && baseProps.denyAllIgwTraffic !== undefined) {
```

Reply:
> Looking at code though, `denyAllIgwTraffic` is never actually given a default value which is why we have this check for undefined here.

The CDK doesn't always set the default values, in fact most of the time CloudFormation does. This is the case here, although it's more complicated than just defaulted to `false`, see [docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_LoadBalancerAttribute.html). The `@default` value needs to be updated to reflect this